Source/Scene/QuadtreePrimitive.js

```diff
@@ -260,7 +262,75 @@ define([
         return object.removeFunc;
     };
 
+    function createTileLoadProgressEventFunction(primitive, loadQueueLength) {
+        return function () {
+            primitive._tileLoadProgressEvent.raiseEvent(loadQueueLength);
+        };
+    }
+
     /**
+     * Checks if the load queue length has changed since the last time we raised a queue change event - if so, raises
+     * a new change event at the end of the render cycle.
+     */
+    function updateTileLoadProgress(primitive, frameState) {
+        var currentLoadQueueLength = primitive._tileLoadQueueHigh.length + primitive._tileLoadQueueMedium.length + primitive._tileLoadQueueLow.length;
+
+        if (currentLoadQueueLength !== primitive._lastTileLoadQueueLength) {
+            frameState.afterRender.push(createTileLoadProgressEventFunction(primitive, currentLoadQueueLength));
+            primitive._lastTileLoadQueueLength = currentLoadQueueLength;
+        }
+
+        var debug = primitive._debug;
+        if (debug.enableDebugOutput  && !debug.suspendLodUpdate) {
+            if (debug.tilesVisited !== debug.lastTilesVisited ||
+                debug.tilesRendered !== debug.lastTilesRendered ||
+                debug.tilesCulled !== debug.lastTilesCulled ||
+                debug.maxDepth !== debug.lastMaxDepth ||
+                debug.tilesWaitingForChildren !== debug.lastTilesWaitingForChildren) {
+
+                console.log('Visited ' + debug.tilesVisited + ', Rendered: ' + debug.tilesRendered + ', Culled: ' + debug.tilesCulled + ', Max Depth: ' + debug.maxDepth + ', Waiting for children: ' + debug.tilesWaitingForChildren);
+
+                debug.lastTilesVisited = debug.tilesVisited;
+                debug.lastTilesRendered = debug.tilesRendered;
+                debug.lastTilesCulled = debug.tilesCulled;
+                debug.lastMaxDepth = debug.maxDepth;
+                debug.lastTilesWaitingForChildren = debug.tilesWaitingForChildren;
+            }
+        }
+    }
+
+    /**
+     * Updates the tile provider imagery and continues to process the tile load queue.
+     * @private
+     */
+    QuadtreePrimitive.prototype.update = function(frameState) {
+        if (defined(this._tileProvider.updateImagery)) {
```

Comment:
The tile provider can be any class that implements the `QuadtreeTileProvider` interface. I didn't add this method to the interface, only to `GlobeSurfaceTileProvider` because I wasn't sure that the imagery needed to be updated in every case.